src/libstd/sys/unix/net.rs

```diff
@@ -66,10 +67,31 @@ impl Socket {
     }
 
     pub fn duplicate(&self) -> io::Result<Socket> {
-        let fd = try!(cvt(unsafe { libc::dup(self.0.raw()) }));
-        let fd = FileDesc::new(fd);
-        fd.set_cloexec();
-        Ok(Socket(fd))
+        use libc::funcs::posix88::fcntl::fcntl;
+        const ZERO: c_int = 0;
```

Comment:
Unless something is known to go wrong, can this just be inlined? If we can't pass literal arguments to variadic functions then we've got bigger fish to fry.